src/compiler.cpp

```diff
@@ -1,69 +1,80 @@
 #include <fstream>
 #include <iostream>
 
-#include "cli.h"
+#include "cli.hpp"
 #include "ast.hpp"
 
 using ast::NodePtr;
 
-NodePtr Parse(const CommandLineArguments& args);
+// Wrapper for ParseAST defined in YACC
+NodePtr Parse(const std::string& compile_source_path);
 
-// Output the pretty print version of what was parsed to the .printed output
-// file.
-void PrettyPrint(const NodePtr& root, const CommandLineArguments& args);
+// Output the pretty print version of what was parsed to the .printed output file.
+void PrettyPrint(const NodePtr& root, const std::string& compile_output_path);
 
-// Compile from the root of the AST and output this to the
-// args.compiledOutputPath file.
-void Compile(const NodePtr& root, const CommandLineArguments& args);
+// Compile from the root of the AST and output this to the compiledOutputPath file.
+void Compile(const NodePtr& root, const std::string& compile_output_path);
 
 int main(int argc, char **argv)
 {
     // Parse CLI arguments to fetch the source file to compile and the path to output to.
     // This retrives [source-file.c] and [dest-file.s], when the compiler is invoked as follows:
     // ./bin/c_compiler -S [source-file.c] -o [dest-file.s]
-    auto command_line_arguments = ParseCommandLineArgs(argc, argv);
+    const auto [compile_source_path, compile_output_path] = ParseCommandLineArgs(argc, argv);
```

Comment:
I see your point, but isn't that why we provide meaningful names (`compile_source_path`, `compile_output_path`), so that our intention is clear? In other words, the header behaves in the safest way by returning named variables boundled in a struct and then the users (us in `compiler.cpp`) choose to use structured binding for convenience of not having to call functions with `command_line_arguments.compile_source_path` etc.?

And from a learning point of view, that simple addition could show students a cool feature that is rather straight forward to understand and embraced in modern C++.

Reply:
I do agree with @Jpnock that in this case I feel as though structured bindings aren't _the most_ suitable, but still think that overall it's worth it to introduce this feature to students.